### PR TITLE
Remove support of annotations

### PR DESCRIPTION
**Pull request by franmomu**

### CODE REVIEW

File: Resources/config/mongodb.xml [hunk 1]
```diff
@@ -112,11 +111,6 @@
 
         <!-- metadata -->
         <service id="doctrine_mongodb.odm.metadata.chain" class="%doctrine_mongodb.odm.metadata.driver_chain.class%" />
-        <service id="doctrine_mongodb.odm.metadata.annotation" class="%doctrine_mongodb.odm.metadata.annotation.class%">
-            <argument type="service" id="doctrine_mongodb.odm.metadata.annotation_reader" />
-            <argument>%doctrine_mongodb.odm.document_dirs%</argument>
-        </service>
-        <service id="doctrine_mongodb.odm.metadata.annotation_reader" alias="annotation_reader" />
         <service id="doctrine_mongodb.odm.metadata.attribute" class="%doctrine_mongodb.odm.metadata.attribute.class%">
```
Comment (franmomu) [inline Resources/config/mongodb.xml:114]: I'm wondering if these services are necessary, I'll try to remove them in another PR

File: DependencyInjection/DoctrineMongoDBExtension.php [hunk 2]
```diff
@@ -479,9 +479,8 @@ private function normalizeDriverOptions(array $connection): array
      *     mappings:
      *         MyBundle1: ~
      *         MyBundle2: xml
-     *         MyBundle3: { type: annotation, dir: Documents/ }
```
Comment (GromNaN) [inline DependencyInjection/DoctrineMongoDBExtension.php:482]: Is there a necessity for `type: attribute` to be in examples? Or is it the default?

**Review by GromNaN — APPROVED** [review]
LGTM

File: DependencyInjection/DoctrineMongoDBExtension.php (hunk 2 above)
Comment (franmomu) [reply DependencyInjection/DoctrineMongoDBExtension.php:482]: Good catch, I've added the example 👍

**Pull request merged**